frontend/pkg/frontend/routes.go

```diff
@@ -102,17 +102,29 @@ func (f *Frontend) routes(r prometheus.Registerer) *MiddlewareMux {
 		MuxPattern(http.MethodGet, PatternSubscriptions, PatternProviders, PatternLocations, api.ClusterVersionTypeName),
 		postMuxMiddleware.HandlerFunc(f.ArmResourceList))
 
-	// Resource ID endpoints
-	// Request context holds an azcorearm.ResourceID
+	// Resource read endpoints
 	postMuxMiddleware = NewMiddleware(
 		MiddlewareResourceID,
 		MiddlewareLoggingPostMux,
 		MiddlewareValidateAPIVersion,
-		MiddlewareLockSubscription,
 		MiddlewareValidateSubscriptionState)
 	mux.Handle(
 		MuxPattern(http.MethodGet, PatternSubscriptions, PatternResourceGroups, PatternProviders, PatternClusters),
 		postMuxMiddleware.HandlerFunc(f.ArmResourceRead))
+	mux.Handle(
+		MuxPattern(http.MethodGet, PatternSubscriptions, PatternResourceGroups, PatternProviders, PatternClusters, PatternNodePools),
+		postMuxMiddleware.HandlerFunc(f.ArmResourceRead))
+	mux.Handle(
+		MuxPattern(http.MethodGet, PatternSubscriptions, PatternProviders, PatternLocations, PatternVersions),
+		postMuxMiddleware.HandlerFunc(f.ArmResourceRead))
+
+	// Resource create/update/delete endpoints
+	postMuxMiddleware = NewMiddleware(
+		MiddlewareResourceID,
+		MiddlewareLoggingPostMux,
+		MiddlewareValidateAPIVersion,
+		MiddlewareLockSubscription,
+		MiddlewareValidateSubscriptionState)
```

Comment:
fwiw, I think future-us should find a factorization that makes it easier to reason about, "did I add the right stuff to this endpoint", but this is an improvement.